CONTRIBUTING.md

```diff
@@ -10,6 +10,29 @@ They are not set in stone.
 Please suggest changes or add other important guidelines.
 Be pragmatic, all rules have exceptions given good reasons.
 
+## Local development
+
+### Local development
+
+1. Clone mockito-release-tools repo, make your changes, then run ```./gradlew fastInstall```
+This will install the artifacts in local maven repository for easy sharing.
+Notice the version you're building in the build output.
+2. Clone mockito-release-tools-example repo and ensure that 'release-tools-example/build.gradle' file uses the correct version of mockito-release-tools (declared at the top of build.gradle).
+It should use the same version that was built in the previous step.
+3. Basic testing (for most contributors):
+ - Smoke test (no tasks are run): ```./gradlew testRelease -m```
+ - Test most things, without actually making any code pushes/publications: ```./gradlew testRelease -x gitPush -x bintrayUpload```
+ - Release notes content: ```./gradlew previewReleaseNotes```
+4. Advanced testing (occasionally, for core developers, edge cases):
+ - Release notes in file: ```./gradlew previewReleaseNotes```, then inspect updated file
+ - Test release needed task: ```./gradlew assertReleaseNeeded```
+ - If you are one of the core developers you can export env variables and even test git push and bintray upload.
+ Run ```./gradlew testRelease``` follow the prompts and export necessary env variables.
+ - Test ciReleasePrepare task but beware that it will reconfigure your working copy!
+ For example, it will switch git user.
+ Remember to switch it back when you finished.
+ - Clone "mockito/mockito" repository and make testing with a proper project :)
```

Comment:
Good point! I need to document what to expect and how to clean up the local working copy when things go wrong. Normally, I don't run clean up tasks, they are automatically run during 'testRelease'.

BTW. at the moment all tasks are documented in ```./gradlew tasks``` and I don't intend to include information what those tasks do in the readme. In the readme/contributing guide I plan to keep use case - based information, e.g. when to use the tasks. I will actually also document ```./gradlew tasks```.

Thank you for review!